pop-miners/altchain-pop-miner/src/main/kotlin/org/veriblock/miners/pop/service/AltchainPopMinerService.kt

```diff
@@ -203,12 +204,11 @@ class AltchainPopMinerService(
             }
         }
 
-        val currentOperations = getOperations().count {
-            it.chain.id == chain.id && it.state.id > MiningOperationState.FAILED.id && it.state.id < ApmOperationState.SUBMITTED_POP_DATA.id
-        }
+        val lastOperationTime = getOperations().maxBy { it.createdAt }?.createdAt
+        val currentTime = LocalDateTime.now()
 
-        if (currentOperations >= 20) {
-            throw MineException("There are 20 other operations in progress! Please, wait for at least one of them to submit the PoP transaction.")
+        if (lastOperationTime != null && lastOperationTime.plusSeconds(1).isAfter(currentTime)) {
```

Comment:
```suggestion
        if (lastOperationTime != null && currentTime < lastOperationTime.plusSeconds(1)) {
```
I think we can do this in kt